Protos/V1/submitter_service.proto

```diff
@@ -149,4 +159,42 @@ message WaitRequest {
 	TaskFilter filter = 1;
 	bool stop_on_first_task_error = 2;
 	bool stop_on_first_task_cancellation = 3;
-}
\ No newline at end of file
+}
+
+/**
+ * Request to watch result states
+ * It contains the list of result ids you want to watch
+ *   and some options to filter out some events
+ */
+message WatchResultRequest {
+  message Options {
+    // List of statuses to ignore in the replies
+    repeated ResultStatus.ResultStatus ignore_statuses = 1;
+    // Skip the sending of result states at the beginning
+    optional bool skip_current_state = 2;
+  }
+  // List of result ids to watch
+  repeated string result_ids = 1;
+
+  // Options
+  optional Options options = 2;
+}
+
+/**
+ * List of Result statuses
+ */
+message WatchResultStream {
+  // Key-Value pair for result id and its status
+  message ResultIdStatus {
+    // Id of the result
+    string result_id = 1;
+    // Status of the result
+    ResultStatus.ResultStatus status = 2;
+  }
+
+  // List of result ids and their statuses
+  repeated ResultIdStatus statuses = 1;
+  // List of ids that were not found
+  // Required in stream mode
+  repeated string not_found_ids = 2;
```

Comment:
Not fan of having this field defined in each message. Not sure whether a oneof would be a good idea here...
In case the message needs to be chuncked, the protocol could ensure that all not_found_ids are sent before the first ResultIdStatus...

Reply:
According to our last conversation, an empty list is the same as the field not being here at all. So I don't really see the difference. Of course, in practice, the not_found_ids will only be sent at the beginning because it will not be retried.

Reply:
> Of course, in practice, the not_found_ids will only be sent at the beginning because it will not be retried.
Shouldn't this be enforced ?

Reply:
Documented? definitely. Enforced? no.

Reply:
explicited